tests/ui/must_use_self_return.rs

```diff
@@ -0,0 +1,42 @@
+#![deny(clippy::must_use_self_return)]
+#![crate_type = "lib"]
+
+pub struct Bar;
+
+pub trait Whatever {
+    // There should be no warning here!
```

Comment:
Why not? That's basically like the `Clone` trait :)

Reply:
Because I only want it on the implementation, not on the "declaration".

Reply:
I think the attribute only has an effect on the declaration in traits, not on implementations. Or am I confused?

Reply:
No, you are perfectly right, I'll change this condition then.